src/Reflection/Adapter/ReflectionClass.php

```diff
@@ -184,7 +184,7 @@ public function getProperty($name)
         $betterReflectionProperty = $this->betterReflectionClass->getProperty($name);
 
         if (null === $betterReflectionProperty) {
-            return null;
+            throw new CoreReflectionException(\sprintf('Property "%s" does not exist', $name));
```

Comment:
Return type hint should also be enforced, I suppose

Reply:
There are currently no return type hints in adapters.